includes/wikia/tests/UserTest.php

```diff
@@ -1,24 +1,144 @@
+			->build();
+		Injector::setInjector( $container );
+	}
+
+
 	/** @dataProvider localOptionNameProvider */
-	public function testCreateLocalOptionName($input, $expected) {
-		$result = call_user_func_array('User::localToGlobalPropertyName', $input);
-		$this->assertEquals($expected, $result, sprintf("failed with input: [%s]", implode(", ", $input)));
+	public function testCreateLocalOptionName( $input, $expected ) {
+		$result = call_user_func_array( 'User::localToGlobalPropertyName', $input );
+		$this->assertEquals( $expected, $result, sprintf( "failed with input: [%s]", implode( ", ", $input ) ) );
 	}
 
 	function localOptionNameProvider() {
 		global $wgCityId;
 		return array(
 			array(
-				array("foo"), "foo-{$wgCityId}",
+				array( "foo" ), "foo-{$wgCityId}",
 			),
 			array(
-				array("foo", 1), "foo-1",
+				array( "foo", 1 ), "foo-1",
 			),
 			array(
-				array("foo", 1, "_"), "foo_1",
+				array( "foo", 1, "_" ), "foo_1",
 			)
 		);
 	}
+
+	public function testGetGlobalPreferenceWithMockedUserPreferenceService() {
+		$this->mockGlobalVariable( 'wgPreferenceServiceRead', true );
+
+		$preference = 'somepref';
+		$value = 'somevalue';
+
+		$preferences = ( new UserPreferences() )
+			->setGlobalPreference( $preference, $value );
+
+		$this->userPreferenceServiceMock->expects( $this->once() )
+			->method( 'getGlobalPreference' )
+			->with( $this->testUser->getId(), $preference, null, false )
+			->willReturn( $value );
+
+
+		$this->userPreferenceServiceMock->expects( $this->once() )
+			->method( 'getPreferences' )
+			->with( $this->testUser->getId() )
+			->willReturn( $preferences );
+
+		$this->assertEquals( $value, $this->testUser->getGlobalPreference( $preference ) );
+	}
+
+	public function testSetGlobalPreferenceWithMockedUserPreferenceService() {
+		// this has side effects in the database as long as we are in migration mode
+		$this->mockGlobalVariable( 'wgPreferenceServiceShadowWrite', true );
+
+		$preference = 'somepref';
+		$value = 'somevalue';
+
+		$this->userPreferenceServiceMock->expects( $this->once() )
+			->method( 'setGlobalPreference' )
+			->with( $this->testUser->getId(), $preference, $value );
+
+		$this->testUser->setGlobalPreference( $preference, $value );
+	}
+
+	public function testGetLocalPreferenceWithMockedUserPreferenceService() {
+		$this->mockGlobalVariable( 'wgPreferenceServiceRead', true );
+
+		$cityId = 12345;
+		$preference = 'somepref';
+		$value = 'somevalue';
+
+		$this->userPreferenceServiceMock->expects( $this->once() )
+			->method( 'getLocalPreference' )
+			->with( $this->testUser->getId(), $cityId, $preference, null, false )
+			->willReturn( $value );
+
+		$this->assertEquals( $value, $this->testUser->getLocalPreference( $preference, $cityId ) );
+	}
+
+	public function testSetLocalPreferenceWithMockedUserPreferenceService() {
+		// this has side effects in the database as long as we are in migration mode
```

Comment:
Because we are still using `setOptionHelper` it should be writing to `user_properties`.